### PR TITLE
feat: add --fork flag to onboard for remote configuration

### LINKED ISSUE #167: Git checkout should be configration to set up remotes
Right now the clone only sets up the origin, but I often want it to have two:

origin (my fork)
upstream (the project I am comiting to).

lets add a new flag that does this during onboarding

### PR DESCRIPTION
**Pull request by jsturtevant**
Closes #167

## Summary

Adds a `--fork <owner/repo>` option to `rally onboard` that configures git remotes for fork-based workflows:

- **origin** → your fork (the `--fork` value)
- **upstream** → the main project repo (the onboard target)

### `bin/rally.js`
- Added `--fork <owner/repo>` option to the onboard command
- Passes fork value through to `onboard()`

### `lib/onboard.js`
- `parseForkArg(fork)` — validates owner/repo format
- `configureForkRemotes(projectPath, fork, _exec)` — renames origin→upstream, adds fork as origin, fetches
- Stores `fork` field in projects.yaml when provided
- Handles edge case: if upstream already exists, preserves it and only updates origin

### `test/onboard-fork.test.js` — 16 tests
- `parseForkArg`: valid input, dots/underscores, null, invalid formats
- `configureForkRemotes`: rename+add, preserve existing upstream, no-origin error, return value
- Integration: fork stored in projects.yaml, remotes configured, no-fork leaves things unchanged, invalid format error, existing upstream preserved

## Testing
All 16 new tests pass. All 37 existing onboard tests pass unchanged.

### CODE REVIEW

File: test/onboard-fork.test.js [hunk 1]
```diff
@@ -0,0 +1,288 @@
+import { test, describe, beforeEach, afterEach } from 'node:test';
+import assert from 'node:assert/strict';
+import {
+  mkdtempSync, rmSync, existsSync, readFileSync,
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply test/onboard-fork.test.js]: Removed unused `existsSync` import. Fixed in 22788ae.

File: test/onboard-fork.test.js [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+      execFileSync('git', ['-C', repoPath, 'remote', 'add', 'origin', originUrl], { stdio: 'ignore' });
+    }
+    return repoPath;
+  }
+
+  // Use injectable _exec to avoid actual network fetch
+  function makeExec(repoPath) {
+    return (args) => execFileSync('git', args, { cwd: repoPath, encoding: 'utf8', stdio: 'pipe' });
+  }
+
```
[the remark this answers is not in the document]
Comment (jsturtevant) [reply test/onboard-fork.test.js]: Removed `makeExec()` — it was dead code. Fixed in 22788ae.
